### PR TITLE
78 create prodigy app

### LINKED ISSUE #78: Create Prodigy app
Create Prodigy app that presents annotators with answers to common parenting questions. There will be 3 answers to each question, one each from a human / GPT4 / the parenting chatbot (retrieval augmented generation system using GPT4).

### PR DESCRIPTION
**Pull request by RFOxbury**

### CODE REVIEW

**Comment by beingkk** [comment]
Hey @RFOxbury, first off, thanks a lot for getting started with this - it's been super helpful!

I've now pushed the changes to the recipe and an expanded readme with instructions. I've also moved a bunch of code in a "scrap" folder, as I ended up not using it for this PR (which is about setting up the platform).

I've gitignored the scrap folder for now as it was throwing up formatting errors from git commit hooks. I can perhaps still reuse some of it when I'll do some quick analysis of the evaluation results. That's just to say that we still have the code you wrote for eg fetching stuff from s3 or viewing annotation results, but it's at the moment not visible for simplicity purposes.

It'd be great if you can take a look and see if everything makes sense - no need to spend too much time with the review or re-running all the scripts. Simply checking the readme and looking at the changes in the code will also be fine. Thank you!

File: src/genai/parenting_chatbot/prodigy_eval/generate_gpt4_answers.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    system_prompt = MessageTemplate.load(SYSTEM_PROMPT)
+    # Load the questions
+    questions = pd.read_json(path_or_buf=QUESTIONS, lines=True).question.to_list()
+    # Generate answers
+    responses = []
+    for question in questions:
+        prompt = MessageTemplate(role="user", content=question)
+        response = TextGenerator.generate(
+            model="gpt-4",
+            temperature=0.6,
```
Comment (RFOxbury) [inline src/genai/parenting_chatbot/prodigy_eval/generate_gpt4_answers.py:40]: Just curious, why is this the temperature setting?

**Review by RFOxbury — COMMENTED** [review]
Thanks Karlis :) It was really useful for me to see how you progressed `best_answer_recipe.py` into the final app.

File: src/genai/parenting_chatbot/prodigy_eval/generate_gpt4_answers.py (hunk 1 above)
Comment (beingkk) [reply src/genai/parenting_chatbot/prodigy_eval/generate_gpt4_answers.py:40]: Good question - not a really good answer, but Kostas said this value worked well for him in other tasks, which is why I've left it at 0.6.

In principle, as I understand this value would suggest a balance between stable (temperature=0) and more random (max temperature=2) modes of operation, with a preference towards the more stable end.

**Pull request merged**